codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java

```diff
@@ -518,14 +518,25 @@ public void encoderAndDecoderAreClosedOnChannelInactive() throws Exception {
     }
 
     @Test
-    public void writeRstOnNonExistantStreamShouldSucceed() throws Exception {
+    public void writeRstOnNonExistantStreamThatMayHaveExistedShouldSucceed() throws Exception {
         handler = newHandler();
+        when(connection.streamMayHaveExisted(NON_EXISTANT_STREAM_ID)).thenReturn(true);
         when(frameWriter.writeRstStream(eq(ctx), eq(NON_EXISTANT_STREAM_ID),
                                         eq(STREAM_CLOSED.code()), eq(promise))).thenReturn(future);
         handler.resetStream(ctx, NON_EXISTANT_STREAM_ID, STREAM_CLOSED.code(), promise);
         verify(frameWriter).writeRstStream(eq(ctx), eq(NON_EXISTANT_STREAM_ID), eq(STREAM_CLOSED.code()), eq(promise));
     }
 
+    @Test
+    public void writeRstOnNonExistantStreamThatMayNotHaveExistShouldNotSucceed() throws Exception {
+        handler = newHandler();
+        when(connection.streamMayHaveExisted(NON_EXISTANT_STREAM_ID)).thenReturn(false);
+        when(frameWriter.writeRstStream(eq(ctx), eq(NON_EXISTANT_STREAM_ID),
```

Comment:
nit: consider removing this as it shouldn't be necessary because we don't expect it to be called and are validating it won't be called in this test.

Reply:
done.